src/rules/tryceratops/rules/prefer_type_error.rs

```diff
@@ -0,0 +1,153 @@
+use ruff_macros::derive_message_formats;
+use rustpython_ast::{Expr, ExprKind, Stmt, StmtKind};
+
+use crate::ast::types::Range;
+use crate::checkers::ast::Checker;
+use crate::define_violation;
+use crate::fix::Fix;
+use crate::registry::Diagnostic;
+use crate::violation::Violation;
+
+define_violation!(
+    pub struct PreferTypeError;
+);
+impl Violation for PreferTypeError {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("Prefer `TypeError` exception for invalid type")
+    }
+}
+
+// check if a call is checking types
+fn check_type_check(checker: &mut Checker, call: &Expr) -> bool {
+    let call_path = checker.resolve_call_path(call);
+    let ids = vec!["isinstance", "issubclass", "callable"];
+    return call_path.as_ref().map_or(false, |call_path| {
+        ids.iter().any(|&i| *call_path.as_slice() == ["", i])
+    });
+}
+
+// check if the test of an If statements is checking types (e.g. via isinstance)
+fn check_test(checker: &mut Checker, test: &Expr) -> bool {
+    match &test.node {
+        ExprKind::BoolOp { values, .. } => {
+            return values.iter().all(|e| check_test(checker, e));
+        }
+        ExprKind::UnaryOp { operand, .. } => check_test(checker, operand),
+        ExprKind::Call { .. } => check_type_check(checker, test),
+        ExprKind::Compare { .. } => false,
+        _ => false,
+    }
+}
+
+fn get_name(exc: &Expr) -> Option<&Expr> {
+    match &exc.node {
+        ExprKind::Name { .. } => Some(exc),
+        ExprKind::Call { func, .. } => {
+            if let ExprKind::Name { .. } = &func.node {
+                return Some(func);
+            }
+            None
+        }
+        _ => None,
+    }
+}
+
+fn check_raise_type(exc: &Expr) -> bool {
+    let builtin_exceptions = vec![
+        "ArithmeticError",
+        "AssertionError",
+        "AttributeError",
+        "BufferError",
+        "EOFError",
+        "Exception",
+        "ImportError",
+        "LookupError",
+        "MemoryError",
+        "NameError",
+        "ReferenceError",
+        "RuntimeError",
+        "SyntaxError",
+        "SystemError",
+        "ValueError",
+    ];
+
+    match &exc.node {
+        ExprKind::Name { id, .. } => {
+            return builtin_exceptions.contains(&id.as_str());
+        }
+        ExprKind::Call { func, .. } => {
+            if let ExprKind::Name { id, .. } = &func.node {
+                return builtin_exceptions.contains(&id.as_str());
+            }
+            false
+        }
+        _ => false,
+    }
+}
```

Comment:
I think it'd be preferable to instead do:

```rs
checker.resolve_call_path(exc).map_or(false, |call_path| 
    [
        "ArithmeticError",
        "AssertionError",
        "AttributeError",
        "BufferError",
        "EOFError",
        "Exception",
        "ImportError",
        "LookupError",
        "MemoryError",
        "NameError",
        "ReferenceError",
        "RuntimeError",
        "SyntaxError",
        "SystemError",
        "ValueError",
    ].iter().any(|target| call_path.as_slice() == ["", target]))
```

That will also be robust to overridden builtins -- so if a user overrides `ValueError`, this won't be truth-y.

Reply:
Thanks! Incorporated this suggestion

Reply:
(`""` as the first segment in the path is a weird convention we have to indicate built-in.)